src/commons/utils/JavaHelper.ts

```diff
@@ -0,0 +1,145 @@
+import setupJVM, { parseBin } from 'java-slang/dist/jvm';
+import { createModuleProxy, loadCachedFiles } from 'java-slang/dist/jvm/utils/integration';
+import { Context } from 'js-slang';
+import { initModuleContext, loadModuleBundle } from 'js-slang/dist/modules/loader/moduleLoader';
+
+import DisplayBufferService from './DisplayBufferService';
+
+const CDN = 'https://source-academy.github.io/modules/java/java-packages/src/';
+
+export async function javaRun(javaCode: string, context: Context) {
+  let compiled = {};
+
+  let files = {};
+  let buffer: string[] = [];
+
+  const readClassFiles = (path: string) => {
+    let item = files[path as keyof typeof files];
+
+    // not found: attempt to fetch from CDN
+    if (!item && path) {
+      const splits = path.split('/');
+      splits.pop(); // classname.class
+      const pkg = splits.join('_');
+
+      const request = new XMLHttpRequest();
+      request.open('GET', `${CDN}${pkg}.json`, false);
+      request.send(null);
+      if (request.status !== 200) {
+        throw new Error('File not found: ' + path);
+      }
+      const json = JSON.parse(request.responseText);
+      // we might want to cache the files in IndexedDB here
+      files = { ...files, ...json };
+
+      if (!files[path as keyof typeof files]) {
+        throw new Error('File not found: ' + path);
+      }
+
+      item = files[path as keyof typeof files];
+    }
+
+    // convert base64 to classfile object
+    const binaryString = atob(item);
+    const bytes = new Uint8Array(binaryString.length);
+    for (let i = 0; i < binaryString.length; i++) {
+      bytes[i] = binaryString.charCodeAt(i);
+    }
+    return parseBin(new DataView(bytes.buffer));
+  };
+  const loadNatives = (path: string) => {
+    // dynamic load modules
+    if (path.startsWith('modules')) {
+      const module = path.split('/')[1] as string;
+      initModuleContext(module, context, true);
+      const moduleFuncs = loadModuleBundle(module, context);
+      const { proxy } = createModuleProxy(module, moduleFuncs);
+      return Promise.resolve({ default: proxy });
+    }
+
+    return import(`java-slang/dist/jvm/stdlib/${path}.js`).then(m => {
+      return m;
+    });
+  };
```

Comment:
Is this the same as the following:

```suggestion
  const loadNatives = async (path: string) => {
    // dynamic load modules
    if (path.startsWith('modules')) {
      const module = path.split('/')[1] as string;
      initModuleContext(module, context, true);
      const moduleFuncs = loadModuleBundle(module, context);
      const { proxy } = createModuleProxy(module, moduleFuncs);
      return Promise.resolve({ default: proxy });
    }

    return await import(`java-slang/dist/jvm/stdlib/${path}.js`);
  };
```

Reply:
I am guessing the `Promise.resolve` would not be necessary as well?
```js
  const loadNatives = async (path: string) => {
    // dynamic load modules
    if (path.startsWith('modules')) {
      const module = path.split('/')[1] as string;
      initModuleContext(module, context, true);
      const moduleFuncs = loadModuleBundle(module, context);
      const { proxy } = createModuleProxy(module, moduleFuncs);
      return { default: proxy };
    }
    return await import(`java-slang/dist/jvm/stdlib/${path}.js`);
  };
  ```
I have committed this change